primitives/src/lib.rs

```diff
@@ -150,4 +153,49 @@ pub mod v0 {
 			NoLayer,
 		}
 	}
+
+	#[cfg(test)]
+	mod tests {
+		use super::*;
+
+		#[test]
+		fn test_error_u32_conversion_with_all_variants() {
+			let error_variants = vec![
```

Comment:
Minor suggestion, variable is redundant and code can be simplify + kept consistent with other tests. 
```suggestion
	        #[test]
		fn test_error_u32_conversion_with_all_variants() {
			// Test conversion for all Error variants
			vec![
				Error::Other,
				Error::CannotLookup,
				Error::BadOrigin,
				Error::Module { index: 1, error: [2, 3] },
				Error::ConsumerRemaining,
				Error::NoProviders,
				Error::TooManyConsumers,
				Error::Token(TokenError::FundsUnavailable),
				Error::Arithmetic(ArithmeticError::Underflow),
				Error::Transactional(TransactionalError::LimitReached),
				Error::Exhausted,
				Error::Corruption,
				Error::Unavailable,
				Error::RootNotAllowed,
				Error::DecodingFailed,
				Error::Unknown { dispatch_error_index: 1, error_index: 2, error: 3 },
			]
			.into_iter()
			.for_each(|error| {
				let status_code = u32::from(error.clone());
				let decoded_error = Error::from(status_code);
				assert_eq!(decoded_error, error);
			});
		}
```